ethtool: Add ethtool PAUSE support

cathay4t opened the pull request:
Example on `npc sim0` output against netdevsim nic:

```yaml
  ethtool:
    pause:
      rx: true
      tx: true
      auto_negotiate: false
```

Python binding also updated.

 Test case included but disabled for CI as CI system has no netdevsim kernel module

cathay4t commented:
@ffmancera PR is ready for review.

cathay4t commented:
The `netlink-generic` and `netlink-ethtool` crate will be removed once https://github.com/little-dude/netlink they merged them.

ffmancera commented on src/lib/ifaces/mac_vtap.rs:
```diff
@@ -5,8 +5,8 @@ use crate::NisporError;
 use serde_derive::{Deserialize, Serialize};
 
-#[serde(rename_all = "lowercase")]
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
```
I think this change is unrelated, is because of rustfmt? Does it matter now?

cathay4t replied:
The cargo clippy warn me that `serde` is used before derive introduced it.

cathay4t replied:
Moved to https://github.com/nispor/nispor/pull/119

ffmancera commented on src/lib/ifaces/mac_vlan.rs:
```diff
@@ -16,8 +16,8 @@ const MACVLAN_MODE_BRIDGE: u32 = 4;
 const MACVLAN_MODE_SOURCE: u32 = 16;
 
-#[serde(rename_all = "lowercase")]
 #[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
```
Same here..

ffmancera commented on src/python/nispor/ethtool.py:
```diff
@@ -0,0 +1,41 @@
+# Copyright 2020 Red Hat
```
Just a nit, 2021 :-)

ffmancera reviewed (changes requested):
The file `src/netlink-ethtool/pause/message.rs` is something missing here or is this a typo? Please, adjust it.

I am also curious, rust files does not require a copyright statement? Thanks!

cathay4t committed:
ethtool: Add ethtool PAUSE support

Example on `npc sim0` output against netdevsim interface:

```yaml
  ethtool:
    pause:
      rx: true
      tx: true
      auto_negotiate: false
```

Python binding also updated.

Test case included but disabled for CI as CI system has no netdevsim
kernel module.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM, thanks!
